### DISCUSSION TITLE
Mehrere Ergebnisse in Datalinq anzeigen

### DISCUSSION THREAD

**Discussion by ue-black** [e-netze/datalinq-community#7] [General]
Hallo!

Wir haben die Eigentümerabfrage im Datalinq im Testssystem realisiert. Wenn man **ein** Grundstück abfragt funktioniert diese auch:

.../webgis5_api/datalinq/report/p-gdbq@gst@view?**kgnr=92001&kgngstnr=92001-20878/1**

![image](https://github.com/gis-eni/datalinq/assets/127403099/b6124ff6-3195-4a30-9b50-1dd95f2a4437)

wenn man aber im Atlas **mehrere** Grundstücke abfragt bekommt folgende URL, die dann auch kein Ergebnis liefert.

![image](https://github.com/gis-eni/datalinq/assets/127403099/feb22c96-93be-48a4-92ca-1ee5b00cd138)

.../webgis5_api/datalinq/report/p-gdbq@gst@view?**kgnr=92001;92001;92001&kgngstnr=92001-20878/1;92001-9154/6;92001-20878/2**

![image](https://github.com/gis-eni/datalinq/assets/127403099/3efbfb35-7da3-4b44-939c-1f3e20c6efa1)

Gibt es eine Möglichkeit die Eigentümer von mehreren Grundstücken/Ergebnissen im Datalinq darzustellen? Müsste man da bei den URL-Parameter etwas ändern?

Unsere Ergebnistabelle im Datalinq:

```
        <tbody>            
            @foreach(var record in Model.Records){
                @DLH.BeginGeoElementFor(record, tag: "tr" 
                                            ,htmlAttributes: new {@class="marker" } 
                                            ,popup: 
                                            "<b>KG-GstNr: </b>" + @record["kgngstnr"] + "<b><br />Einlagezahl: </b>" + @record["ez"] + "<b><br />Eigentümer: </b>" + @record["eigentuemer"]
                                            )
                <td class="gdb-table-colume">@record["kgnr"]</td>
                <td >@record["grstnr"]</td>                
                <td class="gdb-table-colume">@record["ez"]</td>
                <td>@record["eigentuemer"]</td>
                <td>@record["anteil"]</td>
                <td style="text-align: right">@record["gdbflaeche"]</td>
                <td class="gdb-table-colume">@record["geschaeftsfallnummer"]</td>                                
              @*  <td>@record["_location"]</td> *@                
            @DLH.EndGeoElement(tag: "tr")
            }
        </tbody>
```

Vielen Dank für die Hilfe!

**Comment by gis-eni** [comment]
Hallo Klaus!

Mehrere Ergebnisse für die GDB sind sicher schwieriger. Du musst hier deine `Query` so anpassen, dass die mit mehreren Ergebnissen umgehen kann.
Wie deine Abfrage aufgebaut ist, kann ich dir jetzt leider nicht sagen.

Man kann beispielsweise anstatt von ``KGGNR = 92001-101/1`` ein ``KGGNR in (.....,....,.....)``

Wenn man mehrere Parameter übergibt wird es schwieriger. Eventuell kann man dafür eine *Databank Procedure/Funktion* bauen.

Wenn man die `Query` nicht angreifen möchte, kann die Parameter auch in der `Viewer` abfragen, dann `splitten` und drüber iterieren
und dann die einzelnen Ergebnisse als ``SubView```einbinden.
Wenn du dann aber eine Karte auch noch anzeigen willst, wird es natürlich wieder komplieziert.

Wir haben die GDB Abfrage selber nur für ein Grundstück implementiert. Vielleicht Diskussion in die Runde: Wer hat die GDB Abfrage für mehrere Grundstücke im DataLinq implementiert?

LG

**Comment by ue-black** [comment]
Hallo!

Wir haben eine Lösung für unser Thema gefunden. Wir haben folgende Subquery in unsere Abfrage eingebaut:

_#if kgngstnr
       and kgngstnr IN (SELECT unnest(string_to_array(@kgngstnr, ';')))
#endif_    

Das löst das Problem, wenn vom Atlas mehrere Grundstücksnummern geliefert werden, welche mit Strichpunkt getrennt werden: z.B.
![image](https://github.com/gis-eni/datalinq/assets/127403099/6743dc77-ba18-48e1-83ac-4cb9e0d894be)

**Reply by gis-eni** [reply]
Gute Lösung! Wäre ich jetzt nicht drauf gekommen. Kenn ja auch euer Datenmodell nicht... Aber hilft sicher einigen, weil diese Anforderung beim Aufruf aus dem WebGIS immer wieder einmal auftreten wird.
